linera-core/src/remote_node.rs

```diff
@@ -169,14 +169,10 @@ impl<N: ValidatorNode> RemoteNode<N> {
         limit: u64,
     ) -> Result<Vec<ConfirmedBlockCertificate>, NodeError> {
         tracing::debug!(name = ?self.public_key, ?chain_id, ?start, ?limit, "Querying certificates");
-        let range = BlockHeightRange {
-            start,
-            limit: Some(limit),
-        };
-        let query = ChainInfoQuery::new(chain_id).with_sent_certificate_hashes_in_range(range);
-        let info = self.handle_chain_info_query(query).await?;
-        self.node
-            .download_certificates(info.requested_sent_certificate_hashes)
```

Comment:
I think this downloaded at most 1000 at a time (see `download_certificates::download_certificates`). We should do something similar when downloading by heights.

Reply:
Done in [Add loop and early exit to download_certs_at_heights](https://github.com/linera-io/linera-protocol/pull/4373/commits/19e2145238a9094d1819a9fb3777d5ba00106573)